src/librustdoc/core.rs

```diff
@@ -372,7 +377,33 @@ pub fn run_core(options: RustdocOptions) -> (clean::Crate, RenderInfo, RenderOpt
         crate_name,
         lint_caps,
         register_lints: None,
-        override_queries: None,
+        override_queries: Some(|_sess, local_providers, external_providers| {
+            lazy_static! {
+                static ref EMPTY_SET: FxHashSet<LocalDefId> = FxHashSet::default();
+            }
+            // Most lints will require typechecking, so just don't run them.
+            local_providers.lint_mod = |_, _| {};
+            external_providers.lint_mod = |_, _| {};
+            local_providers.typeck_item_bodies = |_, _| {};
```

Comment:
This needs a comment that it prevents `rustc_typeck::check_crate` from calling `typeck_tables_of` on all bodies.